tests/api/test_dnf_callback.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import dnf
+
+from .common import TestCase
+
+
+class DnfCallbackApiTest(TestCase):
+    def test_pkg_downgrade(self):
+        # dnf.callback.PKG_DOWNGRADE
+        self.assertHasAttr(dnf.callback, "PKG_DOWNGRADE")
+        self.assertHasType(dnf.callback.PKG_DOWNGRADE, int)
+
+    def test_downgraded(self):
+        # dnf.callback.PKG_DOWNGRADED
+        self.assertHasAttr(dnf.callback, "PKG_DOWNGRADED")
+        self.assertHasType(dnf.callback.PKG_DOWNGRADED, int)
+
+    def test_pkg_install(self):
+        # dnf.callback.PKG_INSTALL
+        self.assertHasAttr(dnf.callback, "PKG_INSTALL")
+        self.assertHasType(dnf.callback.PKG_INSTALL, int)
+
+    def test_pkg_obsolete(self):
+        # dnf.callback.PKG_OBSOLETE
+        self.assertHasAttr(dnf.callback, "PKG_OBSOLETE")
+        self.assertHasType(dnf.callback.PKG_OBSOLETE, int)
+
+    def test_pkg_obsoleted(self):
+        # dnf.callback.PKG_OBSOLETED
+        self.assertHasAttr(dnf.callback, "PKG_OBSOLETED")
+        self.assertHasType(dnf.callback.PKG_OBSOLETED, int)
+
+    def test_pkg_reinstall(self):
+        # dnf.callback.PKG_REINSTALL
+        self.assertHasAttr(dnf.callback, "PKG_REINSTALL")
+        self.assertHasType(dnf.callback.PKG_REINSTALL, int)
+
+    def test_pkg_reinstalled(self):
+        # dnf.callback.PKG_REINSTALLED
+        self.assertHasAttr(dnf.callback, "PKG_REINSTALLED")
+        self.assertHasType(dnf.callback.PKG_REINSTALLED, int)
+
+    def test_pkg_remove(self):
+        # dnf.callback.PKG_REMOVE
+        self.assertHasAttr(dnf.callback, "PKG_REMOVE")
+        self.assertHasType(dnf.callback.PKG_REMOVE, int)
+
+    def test_pkg_upgrade(self):
+        # dnf.callback.PKG_UPGRADE
+        self.assertHasAttr(dnf.callback, "PKG_UPGRADE")
+        self.assertHasType(dnf.callback.PKG_UPGRADE, int)
+
+    def test_pkg_upgraded(self):
+        # dnf.callback.PKG_UPGRADED
+        self.assertHasAttr(dnf.callback, "PKG_UPGRADED")
+        self.assertHasType(dnf.callback.PKG_UPGRADED, int)
+
+    def test_pkg_cleanup(self):
+        # dnf.callback.PKG_CLEANUP
+        self.assertHasAttr(dnf.callback, "PKG_CLEANUP")
+        self.assertHasType(dnf.callback.PKG_CLEANUP, int)
+
```

Comment:
I think PKG_VERIFY is missing a test here.

Reply:
Thanks, fixed.